File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.py
def solution(id_list, report, k):
    report = set(report)
    dic = {i:{"who":set(), "mail":0} for i in id_list}
    for i in report:
        i = i.split()
        dic[i[1]]["who"].add(i[0])
        
    for i in dic:
        if len(dic[i]["who"])>=k:
            for who in dic[i]["who"]:
                dic[who]["mail"]+=1
    
    answer = [i["mail"] for i in dic.values()]
    return answer

File: 프로그래머스/1/92334. 신고 결과 받기/신고 결과 받기.java
import java.util.*;

class Solution {
    public int[] solution(String[] id_list, String[] report, int k) {
        int[] answer = new int[id_list.length];
        ArrayList<String> list = new ArrayList<>(Arrays.asList(id_list));
        
        HashMap<String, HashSet<String>> map = new HashMap<>();
        for (String r : report) {
            String[] lst = r.split(" ");
            String per1 = lst[0];
            String per2 = lst[1];
            map.putIfAbsent(per2, new HashSet<>());
            map.get(per2).add(per1);
        }
        
        for (String reported : map.keySet()) {
            HashSet<String> reporters = map.get(reported);
            if (reporters.size() >= k) {
                for (String reporter : reporters) {
                    answer[list.indexOf(reporter)]++;
                }
            }
        }
        return answer;
    }
}